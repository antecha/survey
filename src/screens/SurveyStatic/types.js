// @flow
export type { Element } from 'react';

export type SurveyStaticType = {
  value: number,
};
