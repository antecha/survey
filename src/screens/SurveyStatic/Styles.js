import styled from 'styled-components';
import { Radio } from 'antd';

export const StyledRadio = styled(Radio)`
  height: '30px';
  line-height: '30px';
`;

export const QuestionStyle = styled('div')`
  margin-bottom: 10px;
  margin-top: 30px;
`;
