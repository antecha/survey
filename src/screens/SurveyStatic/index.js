// @flow
import React, { Component } from 'react';
import { Radio, Slider, Input, DatePicker, Select, Switch, Button } from 'antd';
import type { SurveyStaticType } from './types';
import type { Element } from './types';
import { StyledRadio } from './styles';
import { StyledQuestion } from './styles';

const RadioGroup = Radio.Group;
const { Option } = Select;

class SurveyStatic extends Component<{}, SurveyStaticType> {
  handleChangeRadio = e => {
    console.log('Changed Radio', e);
  };

  handleChangeInput = e => {
    console.log('Changed Input', e);
  };

  handleChangeSlider = value => {
    console.log('Changed Slider', value);
  };

  handleChangeDate = value => {
    console.log('Changed Date', value);
  };

  handleChangeSwitch = (checked, event) => {
    console.log('Switched', checked, event);
  };
  handleChangeSelect = (value, option) => {
    console.log('Changed Selected item', value, option);
  };
  render(): Element<'div'> {
    return (
      <div>
        <StyledQuestion>Q1. How do you ask in German 'what is your name' ?</StyledQuestion>
        <RadioGroup onChange={this.handleChangeRadio}>
          <StyledRadio value={1}>Wie geht es dir?</StyledRadio>
          <StyledRadio value={2}>Wie spaet es ist?</StyledRadio>
          <StyledRadio value={'Wie heisst du?'}>Wie heisst du?</StyledRadio>
        </RadioGroup>

        <StyledQuestion>Q2. What is your name?</StyledQuestion>
        <Input onChange={this.handleChangeInput} placeholder="Enter your name" />

        <StyledQuestion>Q3. How many states are in Germany?</StyledQuestion>
        <Slider onChange={this.handleChangeSlider} range defaultValue={[5, 35]} />

        <StyledQuestion>Q4. When is your birthday?</StyledQuestion>
        <DatePicker onChange={this.handleChangeDate} />

        <StyledQuestion>Q5. What kind of coffee do you like?</StyledQuestion>
        <Select onChange={this.handleChangeSelect} defaultValue="freddo">
          <Option value="freddo">freddo</Option>
          <Option value="frape">frape</Option>
          <Option value="capuccino">capuccino</Option>
          <Option value="filter cafe">filter cafe</Option>
          <Option value="latte">latte</Option>
          <Option value="americano">americano</Option>
        </Select>

        <StyledQuestion>Q6. What is your gender?</StyledQuestion>
        <Switch
          onChange={this.handleChangeSwitch}
          checkedChildren="Male"
          unCheckedChildren="Female"
          defaultChecked
        />

        <StyledQuestion>
          <Button type="primary">Submit</Button>
        </StyledQuestion>
      </div>
    );
  }
}

export default SurveyStatic;
