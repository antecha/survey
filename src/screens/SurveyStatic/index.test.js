import React from 'react';
import { create } from 'react-test-renderer';
import SurveyStatic from './index';
import { Radio, Slider, Input, DatePicker, Select, Switch, Button } from 'antd';

describe('SurveyStatic', () => {
  const testRenderer = create(<SurveyStatic />);

  it('renders', () => {
    testRenderer.root.findByType(Radio).props.onChange({ target: { value: 'test' } });
    testRenderer.root.findByType(Input).props.onChange({ target: { value: 'test' } });
    testRenderer.root.findByType(Slider).props.onChange({ value: 'test' });
    testRenderer.root.findByType(DatePicker).props.onChange({ dateString: 'test' });
    testRenderer.root.findByType(Select).props.onChange({ value: 'test' });
    testRenderer.root.findByType(Switch).props.onChange({ checked: 'test' });
    testRenderer.root.findByType(Button).props.onClick({});
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
