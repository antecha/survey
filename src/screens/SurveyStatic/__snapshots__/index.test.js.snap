// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveyStatic renders 1`] = `
<div>
  <div
    className="sc-AxirZ iVctAK"
  >
    Q1. How do you ask in German 'what is your name' ?
  </div>
  <div
    className="ant-radio-group ant-radio-group-outline"
  >
    <label
      className="sc-AxjAm glgMFY ant-radio-wrapper"
    >
      <span
        className="ant-radio"
        style={Object {}}
      >
        <input
          checked={false}
          className="ant-radio-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
          value={1}
        />
        <span
          className="ant-radio-inner"
        />
      </span>
      <span>
        Wie geht es dir?
      </span>
    </label>
    <label
      className="sc-AxjAm glgMFY ant-radio-wrapper"
    >
      <span
        className="ant-radio"
        style={Object {}}
      >
        <input
          checked={false}
          className="ant-radio-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
          value={2}
        />
        <span
          className="ant-radio-inner"
        />
      </span>
      <span>
        Wie spaet es ist?
      </span>
    </label>
    <label
      className="sc-AxjAm glgMFY ant-radio-wrapper"
    >
      <span
        className="ant-radio"
        style={Object {}}
      >
        <input
          checked={false}
          className="ant-radio-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="radio"
          value="Wie heisst du?"
        />
        <span
          className="ant-radio-inner"
        />
      </span>
      <span>
        Wie heisst du?
      </span>
    </label>
  </div>
  <div
    className="sc-AxirZ iVctAK"
  >
    Q2. What is your name?
  </div>
  <input
    className="ant-input"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Enter your name"
    type="text"
    value=""
  />
  <div
    className="sc-AxirZ iVctAK"
  >
    Q3. How many states are in Germany?
  </div>
  <div
    className="ant-slider"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="ant-slider-rail"
      style={Object {}}
    />
    <div
      className="ant-slider-track ant-slider-track-1"
      style={
        Object {
          "left": "5%",
          "right": "auto",
          "width": "30%",
        }
      }
    />
    <div
      className="ant-slider-step"
    />
    <div
      aria-disabled={false}
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={5}
      className="ant-slider-handle ant-slider-handle-1"
      onBlur={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="slider"
      style={
        Object {
          "left": "5%",
          "right": "auto",
          "transform": "translateX(-50%)",
        }
      }
      tabIndex={0}
    />
    <div
      aria-disabled={false}
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={35}
      className="ant-slider-handle ant-slider-handle-2"
      onBlur={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="slider"
      style={
        Object {
          "left": "35%",
          "right": "auto",
          "transform": "translateX(-50%)",
        }
      }
      tabIndex={0}
    />
    <div
      className="ant-slider-mark"
    />
  </div>
  <div
    className="sc-AxirZ iVctAK"
  >
    Q4. When is your birthday?
  </div>
  <span
    className="ant-calendar-picker"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={Object {}}
  >
    <div
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
    >
      <input
        className="ant-calendar-picker-input ant-input"
        placeholder="Select date"
        readOnly={true}
        value=""
      />
      <i
        aria-label="icon: calendar"
        className="anticon anticon-calendar ant-calendar-picker-icon"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="calendar"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
          />
        </svg>
      </i>
    </div>
  </span>
  <div
    className="sc-AxirZ iVctAK"
  >
    Q5. What kind of coffee do you like?
  </div>
  <div
    className="ant-select ant-select-enabled"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseOut={[Function]}
    onMouseUp={[Function]}
    onTouchStart={[Function]}
  >
    <div
      aria-autocomplete="list"
      aria-controls="test-uuid"
      aria-expanded={false}
      aria-haspopup="true"
      className="ant-select-selection
            ant-select-selection--single"
      onKeyDown={[Function]}
      role="combobox"
      tabIndex={0}
    >
      <div
        className="ant-select-selection__rendered"
      >
        <div
          className="ant-select-selection-selected-value"
          style={
            Object {
              "display": "block",
              "opacity": 1,
            }
          }
          title="freddo"
        >
          freddo
        </div>
      </div>
      <span
        className="ant-select-arrow"
        onClick={[Function]}
        style={
          Object {
            "WebkitUserSelect": "none",
            "userSelect": "none",
          }
        }
        unselectable="on"
      >
        <i
          aria-label="icon: down"
          className="anticon anticon-down ant-select-arrow-icon"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="down"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
            />
          </svg>
        </i>
      </span>
    </div>
  </div>
  <div
    className="sc-AxirZ iVctAK"
  >
    Q6. What is your gender?
  </div>
  <button
    aria-checked={true}
    className="ant-switch ant-switch-checked"
    defaultChecked={true}
    onChange={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseUp={[Function]}
    role="switch"
    type="button"
  >
    <span
      className="ant-switch-inner"
    >
      Male
    </span>
  </button>
  <div
    className="sc-AxirZ iVctAK"
  >
    <button
      className="ant-btn ant-btn-primary"
      onClick={[Function]}
      type="button"
    >
      <span>
        Submit
      </span>
    </button>
  </div>
</div>
`;
